### PR TITLE
lakefs migrate without kv migrations

### LINKED ISSUE #4179: lakeFS migrate command should be enabled for kv
In order to support data migration for future KV changes, we need to enable 'lakefs migrate' commands.
This is the first version of kv and no active migrations, but the command should be available.

### PR DESCRIPTION
**Pull request by nopcoder**
Fix #4179

### CODE REVIEW

File: cmd/lakefs/cmd/migrate.go [hunk 1]
```diff
@@ -15,23 +18,41 @@ var versionCmd = &cobra.Command{
...
+
+		ctx := cmd.Context()
+		kvParams := cfg.GetKVParams()
+		kvStore, err := kv.Open(ctx, kvParams)
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to open KV store")
+		}
+		defer kvStore.Close()
+
+		version, err := kv.GetDBSchemaVersion(ctx, kvStore)
```
Comment (N-o-Z) [inline cmd/lakefs/cmd/migrate.go:32]: If it's a new installation before setup - maybe we should return a different error on 'not found', such as 'KV not initialized' or something similar?

Comment (N-o-Z) [reply cmd/lakefs/cmd/migrate.go:32]: Perhaps use ValidateSchemaVersion

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Thanks, please see comments

File: cmd/lakefs/cmd/migrate.go (hunk 1 above)
Comment (nopcoder) [reply cmd/lakefs/cmd/migrate.go]: Updated base don modified ValidateSchema.
Will do followup PR to remove additional DB which we no longer use.
And enable retry pg connect, found it missing in the new kv implementation.

File: cmd/lakefs/cmd/migrate.go [hunk 2]
```diff
@@ -18,32 +20,49 @@ var versionCmd = &cobra.Command{
...
-			return
-
-		}
-		logger.WithField("schema_version", version).Info("DB schema version")
+		version := MustValidateSchemaVersion(ctx, kvStore)
 		fmt.Printf("Database schema version: %d\n", version)
 	},
 }
 
+func MustValidateSchemaVersion(ctx context.Context, kvStore kv.Store) int {
```
Comment (N-o-Z) [inline cmd/lakefs/cmd/migrate.go:38]: Does it need to be public?

File: cmd/lakefs/cmd/migrate.go [hunk 3]
```diff
@@ -18,32 +20,49 @@ var versionCmd = &cobra.Command{
...
-		logger := logging.Default()
 		cfg := loadConfig()
-
-		ctx := cmd.Context()
 		kvParams := cfg.GetKVParams()
+		ctx := cmd.Context()
 		kvStore, err := kv.Open(ctx, kvParams)
 		if err != nil {
-			logger.WithError(err).Fatal("Failed to open KV store")
+			_, _ = fmt.Fprintf(os.Stderr, "Failed to open KV store: %s\n", err)
```
Comment (N-o-Z) [inline cmd/lakefs/cmd/migrate.go:28]: Isn't it the same as log.Fatal?

**Review by N-o-Z — APPROVED** [review]
Approved with comments

File: cmd/lakefs/cmd/migrate.go (hunk 3 above)
Comment (nopcoder) [reply cmd/lakefs/cmd/migrate.go:28]: Removed the use of logging for this command.
The logger fatal will print out to the log + exit(1).
The Fprintf will write to std err without and added a call to os.Exit after.
The log output format and file/stderr is based on the lakefs configuration, while the Fprint is not.

File: cmd/lakefs/cmd/migrate.go (hunk 2 above)
Comment (nopcoder) [reply cmd/lakefs/cmd/migrate.go:38]: No - its my default. No one imports cmd package.

Comment (nopcoder) [reply cmd/lakefs/cmd/migrate.go]: Updated to be private

**Pull request merged**